### ISSUE TITLE
[Accessibility] Search sources autocomplete functionality needs to be accessible

### ISSUE THREAD

**Issue by yzen** [devtools-html/debugger.html#4071]
Some background can be found [here]. Keyboard accessibility is pretty great, missing is the semantics:

- [ ] aria-autocomplete="list" must be used in the search [input field][if]
- [ ] aria-controls should be used in the search input to point to the result-list
- [ ] aria-expanded should be used appropriately when the results are shown/hidden
- [ ] result-list should be an aria-live="polite" so screen readers could be notified when new results are rendered
- [ ] result-list should have a role of ['listbox'][lb] and each individual result should have a role of 'option'.
- [ ] aria-activedescendant (set on the listbox) should be pointing to the ID of the currently highlighted result item.
- [ ] Ideally, for each result item the labeling should be as follows: aria-labelledby should be pointing to ".title" child and aria-describedby should be pointing to ".subttitle". This will require assigning ID's to each one of them. This will help with the unnecessary verbosity of each result item where the screen reader would not have to utter a complete path for each one of them.

[here]:  https://www.marcozehe.de/2014/03/11/easy-aria-tip-7-use-listbox-and-option-roles-when-constructing-autocomplete-lists/
[if]: https://www.w3.org/TR/wai-aria/states_and_properties#aria-autocomplete 
[lb]: https://www.w3.org/TR/wai-aria/roles#listbox

### Screenshot

![Screen Shot 2017-09-18 at 8.08.43 PM.png](https://shipusercontent.com/5f0c12dd786fa5deeec037101a48606a/Screen%20Shot%202017-09-18%20at%208.08.43%20PM.png)

**Comment by yzen** [comment]
### aria-autocomplete="list" must be used in the search input field
```diff
diff --git a/src/components/shared/SearchInput.js b/src/components/shared/SearchInput.js
index b8966fa1..4dd1a66c 100644
--- a/src/components/shared/SearchInput.js
+++ b/src/components/shared/SearchInput.js
@@ -118,6 +118,7 @@ class SearchInput extends Component {
       onKeyUp,
       onFocus,
       onBlur,
+      "aria-autocomplete": "list",
       placeholder,
       value: query,
       spellCheck: false,
```

**Comment by yzen** [comment]
### Ideally, for each result item the labeling should be as follows: aria-labelledby should be pointing to ".title" child and aria-describedby should be pointing to ".subttitle". This will require assigning ID's to each one of them. This will help with the unnecessary verbosity of each result item where the screen reader would not have to utter a complete path for each one of them.

```diff
diff --git a/src/components/shared/ResultList.js b/src/components/shared/ResultList.js
index f50bb20e..4edc3d64 100644
--- a/src/components/shared/ResultList.js
+++ b/src/components/shared/ResultList.js
@@ -40,6 +40,8 @@ export default class ResultList extends Component {
       key: `${item.id}${item.value}${index}`,
       ref: index,
       title: item.value,
+      "aria-labelledby": `${item.id}-title`,
+      "aria-describedby": `${item.id}-subtitle`,
       className: classnames("result-item", {
         selected: index === selected
       })
@@ -47,8 +49,8 @@ export default class ResultList extends Component {
 
     return (
       <li {...props}>
-        <div className="title">{item.title}</div>
-        <div className="subtitle">{item.subtitle}</div>
+        <div id={item.id + "-title"} className="title">{item.title}</div>
+        <div id={item.id + "-subtitle"} className="subtitle">{item.subtitle}</div>
       </li>
     );
   }
```

**Comment by tahmidefaz** [comment]
Hello, @RobertHosking, @sanginovs, @davisv007 and I will be working on it. @emmet0r will be helping us out as well.
/claim

**Comment by tahmidefaz** [comment]
\claim

Hello, @RobertHosking, @sanginovs, @davisv007 and I will be working on it. @emmet0r will be helping us out as well.

**Comment by tahmidefaz** [comment]
/claim

Hello, @RobertHosking, @sanginovs, @davisv007 and I will be working on it. @emmet0r will be helping us out as well.

**Comment by jasonLaster** [comment]
Welcome @tahmidefaz, @RobertHosking, @sanginovs, @davisv007!

**Comment by davisv007** [comment]
@jasonLaster We've been able to add most of the attributes from the checklist to the search input and result list. As we test our changes, could you tell me what the kind of behavior we should expect from the screen reader? Everyone in the team is using a distro of linux, which comes with Orca, if that matters. Thank you.

**Comment by yzen** [comment]
@davisv007 You would want to test keyboard interactions with Orca enabled, things like focusing on all focusable (clickable) controls, activating them with Enter/Space. For every change in current screen reader focus, Orca should announce information about what's being focused: type of the control, its name, optionally (depending on the type) value, relation etc.
